packages/components/src/buttons/src/Button.tsx

```diff
@@ -0,0 +1,186 @@
+import { type StyledComponentProps, useStyledSystem, type ResponsiveProp, useResponsiveValue } from "@hopper-ui/styled-system";
+import { type ForwardedRef, forwardRef } from "react";
+import { Button as RACButton, useContextProps, type ButtonProps as RACButtonProps, type PressEvent, composeRenderProps, Provider } from "react-aria-components";
+import styles from "./Button.module.css";
+import { useId } from "react-aria";
+import { cssModule } from "../../utils/src/css-module.ts";
+import { Text } from "../../Text/src/Text.tsx";
+import { composeClassnameRenderProps } from "../../utils/index.ts";
+import { IconContext } from "@hopper-ui/icons";
+import { ButtonContext } from "./ButtonContext.ts";
+import { TextContext } from "../../Text/index.ts";
+import { IconListContext } from "../../IconList/index.ts";
+import { useLocalizedString } from "../../intl/index.ts";
+import { Spinner } from "../../Spinner/index.ts";
+import { isTextOnlyChildren } from "../../utils/src/is-text-only-children.ts";
+
+export const GlobalButtonCssSelector = "hop-Button";
+
+// Won't be needed in next react-aria-components release: https://github.com/adobe/react-spectrum/pull/5850
+const DefaultButtonSlot = "button";
+
+export interface ButtonProps extends StyledComponentProps<RACButtonProps> {
```

Comment:
Igloo also supports buttons that show only the icon when on mobile. Not sure if we want to add something like that too?

Reply:
What is the usecase? This seem everything but accessible.

Reply:
well one example is the Give feeback button here: https://www.figma.com/file/U6Hm6WcMT0eQy9cvSuRcEb/Engagement?type=design&node-id=6728-34679&mode=dev
There are a lot of cases in OV for this. Plus we should always add an aria-label, and maybe even title attributes on buttons for accessibility. sometimes text buttons just won't fit.
It's kin of like when we use icon buttons in Text editors or something, except on desktop, there's text.

Reply:
I was thinking about merging IconButton and Button, but they have different rules in figma. For instance, an IconButton doesn't have an upsell variant.

So we might offer both components, and simply let the user decide which one to show depending on the resolution

Reply:
then that requires the user to have to setup for detecting screen resolution in react (OV does not, they only have the old way of detecting the useAgent) and it could get very repetitive. If we have it in the component, then we can easily do it with css. OR we add something to properly detect the screen resolution so it's easy to use. I think maybe this might be the best and cleanest approach. Especially since we may even need to use it at some point.

Reply:
The styled system already provides everything needed to do this. It's actually quite simple if you want to create a wrapper component for that behavior in OV

function ShowOnlyOnMobileButton(props: ButtonProps & IconButtonProps) {
    const isMobile = useResponsiveValue({base: true, md: false});

    if(isMobile) {
        return <Button {...props} />
    }

    return <IconButton {...props}>
}